### DISCUSSION TITLE
/building-daemon-or-service-app-with-the-microsoft-graph-api/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#206] [Comments]
Building daemon or service app with the Microsoft Graph API
        
[https://www.eliostruyf.com/building-daemon-or-service-app-with-the-microsoft-graph-api/](https://www.eliostruyf.com/building-daemon-or-service-app-with-the-microsoft-graph-api/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Daniel on 2015-11-27 21:55
```

G'day Elio great post. I have been trying to find a solution so I can read from a shared calendar in Office365 and display events from this calendar on a home page of a SharePoint Intranet (same tenant as the shared calendar). Running into authentication problems as apparently accessing a Shared Mailbox uses Basic Auth and using a cross domain rest call from within SharePoint is causing problems..

Would love you thoughts on that and whether this could work in that situation?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2015-12-02 10:23
```

Hi Daniel,

Are you working with a SharePoint on-premises? Are you retrieving an access token in order to retrieve events from the Shared Calendar?

**Reply by comment-migrator** [reply]
```header
Originally posted by Daniel on 2015-12-02 11:29
```

Elio, no not Sharepoint on-presmises it is SharePoint Online and is in the same tenant as the shared calendar.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2015-12-02 16:25
```

How are you trying to retrieve the data?

More information about cors on Office 365: [https://blogs.office.com/20...](https://blogs.office.com/2015/03/06/increasing-opportunities-javascript-developers-office-365-platform/ "https://blogs.office.com/2015/03/06/increasing-opportunities-javascript-developers-office-365-platform/")  
[https://msdn.microsoft.com/...](https://msdn.microsoft.com/en-us/office/office365/howto/create-web-apps-using-CORS-to-access-files-in-Office-365 "https://msdn.microsoft.com/en-us/office/office365/howto/create-web-apps-using-CORS-to-access-files-in-Office-365")

**Comment by comment-migrator** [comment]
```header
Originally posted by Cam Nguyen on 2017-03-31 12:18
```

Hi, can you help me.

I want to build a app only(get Office data without redirect to login page from javascript).  
I have some questions, can you help me.  
1\. Can I get outlook data with tenantId, clientId and clientSecrect(without registing the X509 certificate) or can I use X509 certificate on javascript.  
2\. If user that created app(on Azure Directory) is not a administrator, can we access to view another members meeting.

Thank you.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-03-31 20:50
```

Yes, you can do this via JavaScript / Node.js.

For you first question, all you have to do is make use of the ADAL library.

Here you can find a sample for acquiring your token via a certificate: [https://github.com/estruyf/...](https://github.com/estruyf/SharePoint-Webhook-Node-Sample/blob/master/src/helpers/TokenHelper.ts "https://github.com/estruyf/SharePoint-Webhook-Node-Sample/blob/master/src/helpers/TokenHelper.ts")

Here is a sample were only the clientId and secret is used: [https://github.com/estruyf/...](https://github.com/estruyf/Mail-Tracking-Microsoft-Graph/blob/master/helpers/authHelper.ts "https://github.com/estruyf/Mail-Tracking-Microsoft-Graph/blob/master/helpers/authHelper.ts")

For the second question. If you created the app on Azure and could consent it, I would presume that you have high enough privileges. If not, you will have to ask an admin to give an admin consent to you app.

**Reply by comment-migrator** [reply]
```header
Originally posted by Cam Nguyen on 2017-04-05 04:41
```

Thank you for your reply.  
But I have question about admin role.  
I created a user with role is Global Administrator.  
And then I login as this user and create App.  
But I can't access for get data of all user in this tenant.  
Did I miss setting something. Please help me.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-04-05 13:45
```

Do you want to get the messages with the global administrator account? If that is the case, you can only access your own account via delegated permissions. If you want to be able to get to other users their mail/calendars you will need to create an application which you give application permissions. This works with client + secret / certificate.

**Reply by comment-migrator** [reply]
```header
Originally posted by Cam Nguyen on 2017-04-13 06:12
```

I created app that I can get all member calendar with client Id and secret key(I tried with Postman and I see data) , But now I try with Javascript(not Node.js), I can't call cross domain with POST method(I try to use ajax and also send by iframe). If I send by Iframe, I can receive data, But I can not get access\_token from iframe. Can you help me. Thank you.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-04-20 20:33
```

Are you using a custom API which you built? If so, you probably still have to specify your CORS settings on your site to allow to make calls from the SharePoint domain.

**Reply by comment-migrator** [reply]
```header
Originally posted by Cam Nguyen on 2017-04-26 03:41
```

Thank you so much for your reply. Actually, we use another service so we can't set CORS on server. But we solved this. Thank you again.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-04-26 19:43
```

Great to hear! Thanks for the update. Any information about it you can share?

**Comment by QITSchedular** [comment]
Hi, there I am trying to create an application using react and node js. in the frontend I will have a calender and an email box, any user can come and add his any email and then hit the book button and then I want to create an event in my outlook calender, how can have this email creating calender running on bcakground, since it will run on server I do not want to have any login promp. Please share your thoughts and a sample backend code if possible
